mathlibtools/git_helpers.py

```diff
@@ -0,0 +1,65 @@
+import git   # type: ignore
+from typing import Callable, Iterator, Tuple, List
+
+
+def short_sha(rev: git.Commit) -> str:
+    """ Truncate `rev.hexsha` without ambiguity """
+    return rev.repo.git.rev_parse(rev.hexsha, short=True)
+
+
+def visit_ancestors(rev: git.Commit) -> Iterator[Tuple[git.Commit, Callable]]:
+    r"""
+    Iterate over the ancestors of the commit `rev` in topological order, with
+    the option to prune parents of visited commits.
```

Comment:
I think this is the third time I come here to start reviewing this PR and I stop at the inscrutable docstring. I can vaguely understand what kind of thing you mean by "topological order" (although this is far from explicit) but then I have no idea what the second part means. Reading the example helps to guess, but you should really explain it properly.

Reply:
Thanks for taking a look at it!

Perhaps a reference to the git documentation would be helpful here regarding the meaning I intend by "topological". I'll revise this docstring. Does the example further down the docstring help explain the behavior? I'm happy to spell things out more if you can point out where explanation is most needed.

Reply:
I've rewritten the docstring from scratch - let me know if it's better.

Reply:
This is much better, thanks!